src/tipo_componente.ts

```diff
@@ -1,4 +1,27 @@
 export enum TipoComponente {
   GPU,
-  CPU,
+  CPU
 }
+
+export enum Modelos {
+  RYZEN='RYZEN',
+  INTEL='INTEL',
+  GTX1650='1650',
+  RTX3050='3050',
+  RTX3060='3060',
+  RTX4060='4060',
+  RTX4060TI='4060TI',
+  RTX4070='4070',
+  RTX4070TI='4070TI',
+  RTX4080='4080',
+  RTX4090='4090',
+  RX6600='6600',
+  RX6650XT='6650XT',
+  RX6700XT='6700XT',
+  RX6750XT='6750XT',
+  RX7600='7600',
+  RX7800XT='7800XT',
+  RX7900XT='7900XT',
+  RX7900XTX='7900XTX',
+  UNKNOWN='\0'
```

Comment:
Esto no es demasiado conveniente nunca; tampoco el hecho de que uses un `enun`. No me parece que aporte mucho. En vez de `UNKNOWN` igual debería lanzar una excepción.

Reply:
No consigo verlo de otra forma que hacer un enum. Quiero saber de que modelo es cada componete.

Reply:
El problema es que cada vez que aparezca un modelo nuevo tienes que añadir nuevos elementos al enum; recuerda el principio open/closed: cerrado para modificación, abierto para extensión. Si quieres sólo tener en cuenta ciertos modelos, puedes definir un conjunto o lista que se cargue, por ejemplo. También puedes usar [tipos restringidos](https://stackoverflow.com/questions/61476095/restricting-types-on-object-properties-in-typescript-dynamically-based-on-other), aunque tampoco hace falta.
Piensa en todo caso qué habría que hacer con nuevos modelos. Si la única forma de añadirlos es modificar el código, eso representa una gran cantidad de trabajo; modificar datos es simplemente cuestión de editar un fichero. Por eso os insisto tanto que en la modelización creéis issues, porque este es el tipo de problemas que se plantean y se pueden resolver fácilmente con el código. También es cierto que es del tipo de cosas que igual, no sé, alguna asignatura de programación tendría que enseñar...